File: fxday.py
# Daily update of FX quotes
# peter.gruber@usi.ch, 2024-03

# Requires --  pip install pandas yfinance
import pandas as pd
import yfinance as yf
from datetime import datetime

# Fetch data for EUR/USD currency pair
eur_usd = yf.download("EURUSD=X", period="1d")

# Extract the latest closing price
eur_usd_latest = eur_usd['Close'].iloc[-1]

# Create a DataFrame to store the date and the latest closing price
fx_data = pd.DataFrame({
    'Date': [datetime.now().strftime('%Y-%m-%d')],
    'USD_Latest': [eur_usd_latest]
})

# Append the data to a CSV file, without headers and index
fx_data.to_csv('fx.csv', mode='a', header=False, index=False, sep=",")

File: fxday.R
# Download FX quotes and write to a CSV file
# Programming in Finance and Economics II
# peter.gruber@usi.ch, 2024-03

# To run, use
# Rscript fxday.R
# Then set up a cron job using
# crontab -e

require('quantmod')

USD<-getSymbols("EURUSD=X", auto.assign=F)
USD_latest <- tail(USD$`EURUSD=X.Close`,1)
FXdata <- data.frame(Sys.Date(),USD_latest)

write.table(FXdata,file='fx.csv', sep=",",
            row.names=F, col.names=F, append=T)




